### PR TITLE
Migrate to Develocity Gradle Plugin

### PR DESCRIPTION
**Pull request by 3flex**

### CODE REVIEW

**Comment by 3flex** [comment]
@detekt/maintainers the environment variable for authentication is now named `DEVELOCITY_ACCESS_KEY`. I cannot view the existing value for `GRADLE_ENTERPRISE_ACCESS_KEY` in the GitHub org secrets as secrets are write-only, and I don't seem to have access to generate a new one at https://ge.detekt.dev.

Can someone please generate a new key and copy it to DEVELOCITY_ACCESS_KEY and GRADLE_ENTERPRISE_ACCESS_KEY? We'll need the existing environment variable for a while but it can be deleted in future.

**Review by cortinico — APPROVED** [review]

**Comment by 3flex** [comment]
Windows build failure should be fixed by #7119

File: settings.gradle.kts [hunk 1]
```diff
@@ -43,39 +41,31 @@ include("detekt-utils")
...
-            this as BuildScanExtensionWithHiddenFeatures
-            publishIfAuthenticated()
+            publishing.onlyIf { it.isAuthenticated }
         }
 
-        isUploadInBackground = !isCiBuild
-
-        capture {
-            isTaskInputFiles = true
-        }
```
Comment (TWiStErRob) [inline settings.gradle.kts:70]: @3flex I think this has an equivalent, it wasn't removed:
```gradle
develocity {
	buildScan {
		capture { 
			fileFingerprints = true
		}
	}
}
```

Old:
https://docs.gradle.com/enterprise/gradle-plugin/api/com/gradle/scan/plugin/BuildScanCaptureSettings.html#setTaskInputFiles(boolean)
New:
https://docs.gradle.com/enterprise/gradle-plugin/api/com/gradle/develocity/agent/gradle/scan/BuildScanCaptureConfiguration.html#getFileFingerprints()

Comment (3flex) [reply settings.gradle.kts:70]: Yes but it's now enabled by default according to the docs so I removed the explicit config.

Comment (TWiStErRob) [reply settings.gradle.kts:70]: huh, I thought it might be that, but the javadoc says "defaults to false"

Comment (TWiStErRob) [reply settings.gradle.kts:70]: Can you point to the doc sentence that says it's enabled by default? I can't find it.

Comment (3flex) [reply settings.gradle.kts:70]: https://docs.gradle.com/enterprise/gradle-plugin/current/#capturing_file_fingerprints

Comment (TWiStErRob) [reply settings.gradle.kts:70]: Thanks! Raised https://github.com/gradle/gradle/issues/28694

**Pull request merged**